airbyte-integrations/connectors/source-stripe/source_stripe/partition.py

```diff
@@ -0,0 +1,130 @@
+                elif self._page_token_option.inject_into == RequestOptionType.body_data:
+                    request_body_data = request_body_data | next_page_parameter
+                elif self._page_token_option.inject_into == RequestOptionType.body_json:
+                    request_body_json = request_body_json | next_page_parameter
+
+            response = self._requester.send_request(
+                path=path,
+                request_headers=request_headers,
+                request_params=request_params,
+                request_body_data=request_body_data,
+                request_body_json=request_body_json,
+                next_page_token=next_page_token,
+                log_formatter=log_formatter,
+            )
+            if not response:
+                pagination_complete = True
+            else:
+                records = list(self._record_selector.select_records(response, {}))
+                yield from records
+
+                next_page_token = self._paginator.next_page_token(response, records)
+                if not next_page_token:
+                    pagination_complete = True
+
+
+class StripePartitionGenerator(PartitionGenerator):
+    def __init__(self, stream: Stream, message_repository: MessageRepository, requester: PaginatedRequester):
+        self._stream = stream
+        self._message_repository = message_repository
+        self._requester = requester
+
+    def generate(self, sync_mode: SyncMode) -> Iterable[Partition]:
+        for s in self._stream.stream_slices(sync_mode=sync_mode):
+            yield StripePartition(copy.deepcopy(s), self._message_repository, self._requester)
+
+
+class SourcePartition(Partition):
+    def __init__(self, _slice: Mapping[str, Any], message_repository: MessageRepository, requester: PaginatedRequester):
+        self._slice = _slice
+        self._message_repository = message_repository
+        self._requester = requester
+
+    @property
+    def request_parameters(self) -> Optional[Mapping[str, Any]]:
+        return None
+
+    @property
+    def request_headers(self) -> Optional[Mapping[str, Any]]:
+        return None
+
+    @property
+    def request_body_data(self) -> Optional[Union[Mapping[str, Any], str]]:
+        return None
+
+    @property
+    def request_body_json(self) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def read(self) -> Iterable[Record]:
+        for r in self._requester.send_requests(
+            request_params=self.request_parameters,
+            request_headers=self.request_headers,
+            request_body_data=self.request_body_data,
+            request_body_json=self.request_body_json,
+        ):
+            yield Record(r)
+
+    def to_slice(self) -> Optional[Mapping[str, Any]]:
+        return self._slice
+
+    def __hash__(self) -> int:
+        return hash(str(self._slice))
+
+
+class StripePartition(SourcePartition):
+    @property
+    def request_parameters(self) -> Mapping[str, Any]:
```

Comment:
setting these as constructor arguments would remove the need to create 1 class per connector, which would make the codebase easier to maintain

Reply:
Yes, I've already done it in a similar way, I'm finalizing with the unit tests and will update the PR today.